drivers/spi/spi_sifive.c

```diff
@@ -259,7 +259,7 @@ static struct spi_driver_api spi_sifive_api = {
 	}; \
 	static struct spi_sifive_cfg spi_sifive_cfg_##n = { \
 		.base = DT_INST_REG_ADDR_BY_NAME(n, control), \
-		.f_sys = DT_INST_PROP(n, clock_frequency), \
+		.f_sys = SIFIVE_TLCLK_HZ, \
 	}; \
```

Comment:
Hi. In FU740, SPI and other peripherals are not driven by tlclk. This name is not suitable...

Specification of FU740 "7.1 Clocking" said that
`The hfpclkpll generates the clock for peripherals such as SPI, UART, GPIO, I2C, and PWM.`
This hfpclkpll is different PLL from tlclk.

Reply:
Thanks for checking it. Here is a citation from the the same FU740 specification, "19.4 Serial Clock Divisor Register (sckdiv)".

`The relationship between the input clock and SCK is given by the following formula:`
`fsck = (fin) / 2 * (div + 1)`
`The input clock is the bus clock tlclk.`

And, there is a similar note regarding tlclk being the input clock for the UART peripheral (16.9). But I do see that PCLK is specified as the input in Linux (arch/riscv/boot/dts/sifive/fu740-c000.dtsi).

So I suspect you are correct, but it's difficult to resolve these conflicting statements in the specification. I will do some testing on an FU740 board to confirm.

Reply:
I tested on an FU740 board and confirmed that UART is functional with the existing 125125000Hz clock setting. "_The input clock is the bus clock tlclk._" - this is an erroneous statement for FU740. I will revise the patch accordingly.